terraform/files/deploy.sh

```
@@ -0,0 +1,139 @@
+
+# Check if required environment variables are set
+if [ -z "$COHERE_API_KEY" ] || [ -z "$OPENAI_API_KEY" ]; then
+  echo "ERROR: Required API keys are not set!"
+  exit 1
+fi
+
+# Check for permissions and fix if needed
+if [ ! -w "$APP_DIR" ]; then
+  echo "Need to fix permissions on $APP_DIR"
+  sudo mkdir -p "$APP_DIR"
+  sudo chown -R $(whoami):$(whoami) "$APP_DIR"
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)